### ISSUE TITLE
[css-image-animation] alt-less (decorative) images need to be addressed

### ISSUE THREAD

**Issue by aardrian** [w3c/csswg-drafts#13785]
Reading through [§ 5. Accessibility Considerations](https://www.w3.org/TR/css-image-animation-1/#a11y) I see these statements:

> * Web pages can already contain animated images, and appropriate [alt](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-alt) text is already expected to be provided. No change is expected nor necessary to accommodate for pausing/resuming animations. Descriptions like ”Cartoon coyote being squashed by a falling anvil“, “cute dancing hamster”, “under construction”, or “rotating loading indicator” remain equally appropriate and evocative whether the animation is running or not.

It is not uncommon for me to encounter alt-less animated images in accessibility reviews, design systems, or articles (whether by leaving the `alt` attribute blank or using [empty quotes after the `/`](https://www.w3.org/TR/css-content-3/#alt) in images called via CSS generated content). Whether a WCAG violation or not is moot (some animated images can be truly decorative).

Further, [not all screen reader users are blind](https://adrianroselli.com/2017/02/not-all-screen-reader-users-are-blind.html) and still need the ability to pause animated images.

The problem is that once an `<img>` element gets a blank `alt`, the user agent no longer exposes it to screen readers. This is a problem when a screen reader user wants to pause/stop an image since it's not there as far as the browser is concerned.

While the statement above that I quoted from the spec is intended to say (I think) that no changes needs to be made to authored alt text, the reality is that if no change is made then a decorative animated image cannot be stopped. Which means this first bullet under [§ 5. Accessibility Considerations](https://www.w3.org/TR/css-image-animation-1/#a11y) is already incorrect:

> * The features introduced by this specification enable authors to address what would otherwise be a violation of [WCAG 2.2 §  Success Criterion 2.2.2 Pause, Stop, Hide](https://www.w3.org/TR/WCAG22/#pause-stop-hide).

So either urgency is needed for #13784 or some other method to address this has to be developed.

For clarity, when I refer to "decorative" images, I am referring the WAI / WCAG / existing CSS definitions, not the one in this draft. See #13783 for more detail.

**Comment by aardrian** [comment]
I think this HTML issue is related (or at least solving how to expose alt-less images would be): [`<img controls>` guidance for declaration of alt #12377](https://github.com/whatwg/html/issues/12377)

**Comment by frivoal** [comment]
> While the statement above that I quoted from the spec is intended to say (I think) that no changes needs to be made to authored alt text, the reality is that if no change is made then a decorative animated image cannot be stopped.

I think may be missing something. Whether an image has alt text or not doesn't change whether you can stop its animation. This spec does not define an interactive feature that lets users turn image animations on or off, which might be suppressed by an absent alt text. It defines a css property that does so. If the author intends to enable users to turn image animations on and off manually, they need to hook that css property to something the user can interact with. Whatever that thing is, it needs, as anything users interact with, to be made properly accessible. But that's on the author. However, the author could also turn animations on or off based on something else than an explicit user interaction, such as response to media queries, context based…

> Which means this first bullet under [§ 5. Accessibility Considerations](https://www.w3.org/TR/css-image-animation-1/#a11y) is already incorrect:
> 
> > * The features introduced by this specification enable authors to address what would otherwise be a violation of [WCAG 2.2 §  Success Criterion 2.2.2 Pause, Stop, Hide](https://www.w3.org/TR/WCAG22/#pause-stop-hide).

I don't think it is incorrect. It « enables **the author** to address […] », not the user directly (unless the user writes user style.

What am I missing?

**Comment by aardrian** [comment]
> Whether an image has alt text or not doesn't change whether you can stop its animation

No, but it _does_ control whether or not the image is exposed to users — regardless of whether the author adds a feature to control it. If the image isn't exposed, it doesn't matter if the author adds that feature since a user may not be able to get to it to trigger the control written by the author.

> It defines a css property that does so turns image animations on or off.

I think there's a typo in this sentence? I think you mean it defines a CSS property that allows authors access to the state?

>  If the author intends to enable users to turn image animations on and off manually, they need to hook that css property to something the user can interact with. Whatever that thing is, it needs, as anything users interact with, to be made properly accessible.

Agreed, but the image that the authored feature controls also needs to be exposed. Otherwise it's like having a play button but no corresponding video on the page (or more play buttons than there are videos, which can make it particularly daunting to understand which controls which).

My suggestion is essentially that non-blank alternative text should be required "to accommodate for pausing/resuming animations".

**Comment by frivoal** [comment]
> > Whether an image has alt text or not doesn't change whether you can stop its animation
> 
> No, but it _does_ control whether or not the image is exposed to users — regardless of whether the author adds a feature to control it. If the image isn't exposed, it doesn't matter if the author adds that feature since a user may not be able to get to it to trigger the control written by the author.

A "turn off all image animations on the page" button would be accessible (if the button is properly designed) and understandable whether or not there are images on the page, animated or not, with alt text or not. It's not terribly useful if there are no images on the page, but that's a separate question.

Let's say you're on a page with animated borders in the style of a spinning barber-pole. There would generally not be an expectation that such images would have alt text, they're purely decorative. I don't see why we should require that alt text must be added just because the author wants to add a "stop border animations" button on the page.

> I think there's a typo in this sentence? 

fixed.
